### ISSUE TITLE
New Rule: Do not use private autogenerated properties

### ISSUE THREAD

**Issue by pavel-mikula-sonarsource** [SonarSource/sonar-dotnet#5254]
As a result of community thread: https://community.sonarsource.com/t/false-positive-s4275-getters-and-setters-should-access-the-expected-fields/54425/10

The rule should detect private autogenerated properties and suggest using a field instead. 
Code fix should be easy to do.
SonarWay

We should avoid raising issues when:
* Property is decorated with an attribute - could be used for some kind of serialization
* Property implements interface

```
Public Interface INoisy

    Property Count As Integer

End Interface

Public Class Collection
    Implements INoisy

    Private Property Count As Integer Implements INoisy.Count	'Compliant

End Class
```

**Comment by pavel-mikula-sonarsource** [comment]
Internal ticket [NET-1870](https://sonarsource.atlassian.net/browse/NET-1870)

[NET-1870]: https://sonarsource.atlassian.net/browse/NET-1870?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
